File: string__remove_duplicates.py
def toList(str):
    chars = []
    for c in str:
        chars.append(c)
    return chars

def removeDuplicates(list):
    i = j = 1
    while i < len(list):
        if list[i] != list[i - 1]:
            list[j] = list[i]
            j += 1
        i += 1
    return ''.join(list[0:j])

list = toList('ccaafooaa')
list.sort()
print(removeDuplicates(list))

File: string__remove_duplicates.js
console.log(
    Object.keys(
        'foobarbrarorr'
            .split('')
            .reduce((x, y) => { x[y] = 1; return x; }, {})
    )
        .join('')
);

function removeDuplicates(chars) {
    let i = 1,
        j = 1;

    while (i < chars.length) {
        if (chars[i] !== chars[i - 1]) {
            chars[j] = chars[i];
            j++;
        }
        i++;
    }

    return chars.slice(0, j);
}

let chars = 'foobar'.split('');
chars.sort();
console.log(removeDuplicates(chars).join(''));